tests/codebuild/feature_tests.sh

```diff
@@ -76,17 +118,57 @@ function verify_resource_has_additional
 # This function verifies that each Failed trainingjob for a given HPO job have
 # the additional field in their status set.
 # Parameter:
-#    $1: Instance of HyperParameterTuningJob
+#    $1: Job namespace
+#    $2: Instance of HyperParameterTuningJob
 function verify_failed_trainingjobs_from_hpo_have_additional
 {
-  local crd_instance="$1"
+  local crd_namespace="$1"
+  local crd_instance="$2"
 
-  local sagemaker_hpo_job_name="$(kubectl get hyperparametertuningjob "$crd_instance" -o json | jq -r '.status.sageMakerHyperParameterTuningJobName')"
+  local sagemaker_hpo_job_name="$(kubectl get -n "$crd_namespace" hyperparametertuningjob "$crd_instance" -o json | jq -r '.status.sageMakerHyperParameterTuningJobName')"
 
   # Loop over every failed training job and get the k8s resource name
-  for training_job in $(kubectl get trainingjob | grep Failed | grep "$sagemaker_hpo_job_name" | awk '{print $1}'); do
-    if ! verify_resource_has_additional TrainingJob "$training_job"; then
+  for training_job in $(kubectl get -n "$crd_namespace" trainingjob | grep Failed | grep "$sagemaker_hpo_job_name" | awk '{print $1}'); do
+    if ! verify_resource_has_additional "$crd_namespace" TrainingJob "$training_job"; then
       return 1
     fi
   done
+}
+
+# This function verifies that a given training job does not have a status.
+# Returns 1 if job creation fails.
+# Parameter:
+#    $1: CRD namespace
+#    $2: CRD type
+#    $3: Instance of CRD
+function verify_trainingjob_has_no_status
+{
+  local crd_namespace="$1"
+  local crd_type="$2"
+  local crd_instance="$3"
+  local sagemaker_training_job_status="$(kubectl get -n "$crd_namespace" "$crd_type" "$crd_instance" -o json | jq -r '.status')"
+
+  if [ "$sagemaker_training_job_status" = null ]; then
+    echo "[SUCCESS] $crd_type deployed to $crd_namespace namespace does not have a status"
+    return 1
+  fi
+}
+
+# This function verifies that a given training job does not have a sagemaker job name. 
+# Returns 1 if job creation fails.
+# Parameter:
+#    $1: CRD namespace
+#    $2: CRD type
+#    $3: Instance of CRD
+function verify_trainingjob_has_no_sagemaker_name
```

Comment:
nit: This function accepts any crd_type not just trainingjob. However since we use `.status.sageMakerTrainingJobName` it should probably only accept trainingjobs.

Reply:
Fixed